core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -848,7 +853,7 @@ private[spark] class AppStatusListener(
     }
 
     stages.foreach { s =>
-      val key = s.id
+      val key = Array(s.info.stageId, s.info.attemptId)
```

Comment:
Use a case class instead?
Or create a function `xxxKey` and return the Array/Tuple. 
Using `Array(s.info.stageId, s.info.attemptId)` in several places looks not robust enough